components/mediation/mediators/dataservices-mediator/org.wso2.micro.integrator.mediator.dataservice/src/main/java/org/wso2/micro/integrator/mediator/dataservice/DataServiceCallMediatorFactory.java

```diff
@@ -202,4 +181,161 @@ private List extractOperations(OMElement operationsElemet, DataServiceCallMediat
+                    if (operationsTypeElement != null) {
+                        handleException("The 'source' type is set to body. Inline configurations " +
+                                "are only applicable for source type 'inline'.");
+                    }
+                }
+            } else {
+                handleException("The 'type' attribute in 'source' element is required for the configuration");
+            }
+        } else {
+            log.warn("The 'source' element is missing in the configuration. The default value will be set to 'body'.");
+            if (operationsTypeElement != null) {
+                handleException("The 'source' type is set to body. Inline configurations are " +
+                        "only applicable for source type 'inline'.");
+                mediator.setSourceType(DataServiceCallMediatorConstants.SOURCE_BODY_TYPE);
+            }
+        }
+        return mediator;
+    }
+
+    /**
+     * Method to set the target element.
+     * @param mediator dataServiceCallMediator instance
+     * @param element OMElement
+     * @return dataServiceCallMediator with configured target element
+     */
+    private DataServiceCallMediator configureTargetElement(DataServiceCallMediator mediator, OMElement element) {
+        OMElement targetElement = element.getFirstChildWithName(TARGET_Q);
+        if (targetElement != null) {
+            OMAttribute typeAtr = targetElement.getAttribute(TYPE_Q);
+            if (typeAtr != null) {
+                String typeValue = typeAtr.getAttributeValue();
+                mediator.setTargetType(typeValue);
+                if (typeValue.equals(DataServiceCallMediatorConstants.PROPERTY)) {
+                    OMAttribute propertyAtr = targetElement.getAttribute(NAME_Q);
+                    if (propertyAtr != null) {
+                        if (!propertyAtr.getAttributeValue().isEmpty()) {
+                            String propertyName = propertyAtr.getAttributeValue();
+                            if (isDynamicName(propertyName)) {
+                                Value targetDynamicName = createDynamicNameValue(propertyName, targetElement);
+                                mediator.setTargetDynamicName(targetDynamicName);
+                            } else {
+                                mediator.setTargetPropertyName(propertyName);
+                            }
+                        } else {
+                            handleException("The 'name' attribute in 'target' element is empty. " +
+                                    "Please enter a value.");
+                        }
+                    } else {
+                        handleException("The 'name' attribute in 'target' element  is missing " +
+                                "in the configuration.");
+                    }
+                }
+            } else {
+                handleException("The 'type' attribute in 'target' element is required for the configuration");
```

Comment:
As there is a default value, shouldn't we add that into configuration with a warning?

Reply:
Fixed in 454c99f9196bd3b43193e344c9079958e9ae0eb1